tests/miral/runner.cpp

```diff
@@ -73,6 +93,73 @@ TEST_F(Runner, x11_socket_is_not_returned_by_default)
     miral::TestServer::invoke_runner([](auto& runner){ EXPECT_FALSE(runner.x11_display().is_set()); });
 }
 
+TEST_F(Runner, register_signal_handler_before_setup_invokes_callback_after_setup)
+{
+    int const signum{SIGUSR1};
+
+    WaitForLockingCallback();
+    register_signal_handler({signum}, [this](int) { LockingCallback(); });
+
+    miral::TestServer::SetUp();
+
+    EXPECT_CALL(*this, callback())
+        .Times(AtLeast(1));
+    kill(getpid(), signum);
+
+    miral::TestServer::TearDown();
+}
```

Comment:
Upon reading this, I thought a separate `wait` outside of TearDown() would be unnecessary. As it turns out, it is. 

The `FdHandle` wasn't being held once it went out of scope after a test run (which then calls `TearDown()` automatically), so the Fd was being dropped and the callback usually wasn't triggered.

Calling `TearDown()` directly "fixed" the problem because it was keeping the handle in scope!